src/actions/publish.js

```diff
@@ -263,9 +263,8 @@ export function attemptToPublish(): ThunkAction<Promise<boolean>> {
       // Because we want to store the published profile even when the upload
       // generation changed, we store the data here, before the state is fully
       // updated, and we'll have to predict the state inside this function.
-      // Note that this function is asynchronous, we don't await it on purpose.
       // We catch all errors in this function.
-      persistJustUploadedProfileInformationToDb(
+      await persistJustUploadedProfileInformationToDb(
```

Comment:
Now it is necessary to wait for this, so that when the hash is changed, the information is also available in the DB and we can fetch it in the new component.

I believe we could also change the cached value as part of the publish operation but I believe that handling it all from the new component makes it nicely separated.